### PR TITLE
Add visibility field to post type controller

### PR DESCRIPTION
**Pull request by spacedmonkey**
See this issue for more information https://github.com/WordPress/gutenberg/issues/34443

Trac ticket: https://core.trac.wordpress.org/ticket/54055

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by TimothyBJacobs** [comment]
We may want to look at `is_post_type_viewable`.

**Comment by spacedmonkey** [comment]
> We may want to look at `is_post_type_viewable`.

This data is already in the endpoint. 

https://github.com/WordPress/wordpress-develop/blob/26c0800c4625e1668bf91f97319abebb9ad6bd8f/src/wp-includes/rest-api/endpoints/class-wp-rest-post-types-controller.php#L197-L199

**Comment by TimothyBJacobs** [comment]
Ah, my mistake. So should we still be exposing `public` and `publicly_queryable` then?

**Comment by peterwilsoncc** [comment]
> So should we still be exposing `public` and `publicly_queryable` then?

`public` should probably not be exposed, it's pretty meaningless.  A post type can be `public` without being viewable by registering a CPT with these arguments:

```php
[
 'public' => true,
 'exclude_from_search' => true,
 'publicly_queryable' => false,
 'show_in_nav_menus' => false,
 'show_ui' => false,
 'show_in_rest' => false,
]
```

`publicly_queryable` is of more use as it's what determines if a post type is viewable for CPTs but going via the helper function `is_post_type_viewable()` is the safest result.

**Comment by spacedmonkey** [comment]
@peterwilsoncc Good catch. I have removed it now.

**Comment by TimothyBJacobs** [comment]
We should be ditching `publicly_queryable` as well then, right? Since we are already exposing the correct computed value?

File: tests/phpunit/tests/rest-api/rest-post-types-controller.php [hunk 1]
```diff
@@ -216,6 +217,11 @@ protected function check_post_type_obj( $context, $post_type_obj, $data, $links
 				$viewable = is_post_type_viewable( $post_type_obj );
 			}
 			$this->assertSame( $viewable, $data['viewable'] );
+			$visibility = array(
+				'show_in_nav_menus' => (bool) $post_type_obj->show_in_nav_menus,
+				'show_ui'           => (bool) $post_type_obj->show_ui,
+			);
+			$this->assertEqualSets( $visibility, $data['visibility'] );
```
Comment (TimothyBJacobs) [inline tests/phpunit/tests/rest-api/rest-post-types-controller.php:224]:
```suggestion
			$this->assertSame( $visibility, $data['visibility'] );
```

**Comment by TimothyBJacobs** [comment]
Merged in https://core.trac.wordpress.org/changeset/51959.

**Pull request closed without merging**